### ISSUE TITLE
[Feature request]: Make Project64 advise on first-run when it doesn't support an OS

### ISSUE THREAD

**Issue by dftf-stu** [project64/project64#2254]
### Request box

When _Project64_ is ran on an unsupported OS, it would be better to show users a clear message advising them of this, and link to the final version they can download which will work.

As an example UI mock-up:

![WinXP_Support_SuggestedMessage](https://user-images.githubusercontent.com/83965343/180011359-b12749af-0018-4fa4-ab7e-ed1c29a7bd65.png)

(If there is anything else they may need to install, such as a particular release of the Microsoft Visual C++ libraries, then this should be mentioned also.)

### Reason box

I have tested the most-recent version of each milestone release (4.0.0.5846; 3.0.1.5669; 3.0.0.5648; 2.4.0.1868; 2.3.2.202; 2.3.0.210) inside a virtual-machine running _Windows XP Home Service Pack 3_ and found that they **all** display the initial "Welcome" dialog:

![WinXP_Support](https://user-images.githubusercontent.com/83965343/180010692-be0e5f6f-7cc8-410e-81c1-08a2734198a5.png)

This will confuse some users into thinking that the app should work on their system, when it is actually not supported, but nothing makes this clear -- and if you double-click the "Project64.exe" file, no error-message appears then, either; the process simply appears for a brief second in _Task Manager_, then disappears.

### Examples

There are many examples of apps which already display clear messages, usually pre-install, to advise _XP_ is a no-go:

**iTunes**
"iTunes requires that your computer is running Windows 10"

**LibreOffice**
"LibreOffice 7.2.7.2 cannot be installed on this Windows version.  Windows 7 SP1 or newer is required."

**qBittorrent**
"This qBittorrent version requires at least Windows 7"

**SumatraPDF**
"Version 3.1.2 is the final version supported on Windows XP.  Setup cannot continue."

**Comment by dftf-stu** [comment]
Just to add: I've done some testing and version 2.3.2 is the last that supports Windows XP and Vista, so such a message should appear for users on both platforms.

Windows 7 SP1, however, as-of-now still supports both the latest 3.0.1.5669 and 4.0.0.5846 Dev releases.

HOWEVER, Microsoft will cease all-support for Windows 7 and Windows 8.1 in January 2023, so I will imagine the _Project64_ team may look to abandon support for those after that date.  If so, please then add such a message into future versions to similarly advise that the latest version is users can use!

**Comment by dftf-stu** [comment]
Another example UI mock-up, this time for Vista:

![ERROR_VISTA](https://user-images.githubusercontent.com/83965343/180069428-0ac3288f-2262-4a6c-a46b-3f441257083f.png)

**Comment by flagrama** [comment]
This doesn't make sense to do. An unsupported operating system will eventually just stop running the application. You'd literally have to create a separate application binary of just this window for Win9x for example.

The installer checking instead does make sense which is what the other applications do, but I don't know if the one pj64 uses supports this check. Project64 isn't _only_ available as an installer though (and sometimes actively discouraged if issues with it are found) and the executable cannot reliably check this and never will.

**Comment by dftf-stu** [comment]
@flagrama: well, the current situation isn't ideal either.

A user on XP can come to this site, and according to the README on the overview page, be advised that XP is the minimum OS supported.  They then click the link to download the most-recent stable version, get greeted by the "Welcome" screen and think all is well, only to then find afterwards the Project64 app doesn't do anything, and there are no error-messages.  Which will leave them trying to troubleshoot something which cannot be solved.

I don't think we need to be silly, and create messages for things like Windows 3.x, Windows 95, Windows 2000 and MS-DOS.  XP and Vista are the only ones I'd suggest this for, as nothing newer than the 2.3.2 release works on them, even though nothing on this site, or the official site for Project64, makes that clear

**Comment by flagrama** [comment]
I'd just update the README and related. Nobody does this because it'd be a nightmare to support, you'd literally be supporting an unsupported OS just to say it's unsupported. A waste of time and effort and extremely limiting.

And this is assuming zilmar doesn't prefer to just "fix" whatever broke XP and Vista.

**Comment by flagrama** [comment]
@dftf-stu Are you on the pj64 discord?

**Comment by DerekTurtleRoe** [comment]
I can update the README if needed, but I would just need to know if this would be either quickly fixed in the latest builds, etc.

I can also add a note that support for specific Windows versions is "limited" meaning if something breaks one commit then is fixed or whatever it isn't necessarily something we will fix ASAP if it isn't a huge priority.

I would be interested to know how many users are on versions of Windows before 7. Windows 8.1 will reach EOL in January, so things could break for some users anytime after then as well.

**Comment by dftf-stu** [comment]
@flagrama: no, cannot stand Discord, sorry... try to avoid social-media where possible.

**Comment by dftf-stu** [comment]
@DerekTurtleRoe: the [README on here](https://github.com/project64/project64/?tab=readme-ov-file#supported-requirements) has been updated at some-point to now say only Windows 10 (64-bit) and 11 are the minimum supported.

I'd suggest it might be good to add a section on there for older versions of Windows, and provide the archive version, and then this ticket (and similar ones about XP and Vista support) could be closed?

For example:

**Using an older version of Windows?**

ℹ You're free to download one of these older versions, but we no-longer offer any support for them:

- Windows 10 (32-bit), 8.1/8.0 or 7: the [3.0.1 stable] release is the last supported
- Windows Vista or XP: the [2.3.2 stable] release is the last supported

We would recommend ensuring the latest Service Pack update (7 SP1, Vista SP2, XP SP3) is installed.
